### PR TITLE
Add NTPServerSync before getNetworkTime

### PR DESCRIPTION
**Pull request by pennam**
Calling `getNetworkTime` without `NTPServerSync` returns some weird time values in the past

`1388591169 -> Wed Jan 01 2014 15:46:09 GMT+0000`

Adding `NTPServerSync` call solves this issue

### CODE REVIEW

**Review by sebromero — CHANGES_REQUESTED** [review]
Let's use the return code of `NTPServerSync`. -1 indicates an error. Maybe we can return UNIX time `Time(1970,1,1,0,0,0,0)` in that case.

File: src/ArduinoCellular.cpp [hunk 1]
```diff
@@ -122,9 +122,16 @@ Time ArduinoCellular::getGPSTime(){
...
-    int year, month, day, hour, minute, second;
+    int year = 1970;
+    int month = 1;
+    int day = 1;
+    int hour = 0;
+    int minute = 0;
+    int second = 0;
     float tz;
-    modem.getNetworkTime(&year, &month, &day, &hour, &minute, &second, &tz);
+    if (modem.NTPServerSync()) {
```
Comment (sebromero) [inline src/ArduinoCellular.cpp:132]:
```suggestion
    if (modem.NTPServerSync() == 0) {
```
I think a return code of 0 indicates success. @cristidragomir97 can you double check?

Comment (pennam) [reply src/ArduinoCellular.cpp]: Oh yes, i was for sure wrong. I've changed it checking if `!= -1`  to be on the safe side :slightly_smiling_face:

Comment (sebromero) [reply src/ArduinoCellular.cpp]: @pennam I think that -1 is the generic error code while there are other `+CNTP: ` result codes that indicate an error. I'd check for == 0.

Comment (pennam) [reply src/ArduinoCellular.cpp]: @sebromero Thanks! I was wrong again :smiling_face_with_tear:

**Review by sebromero — APPROVED** [review]
LGTM

**Pull request merged**